datadog-cloudformation-common-python/src/datadog_cloudformation_common/utils.py

```diff
@@ -21,3 +31,20 @@ def http_to_handler_error_code(http_error_code: int) -> HandlerErrorCode:
     elif http_error_code == 429:
         return HandlerErrorCode.Throttling
     return HandlerErrorCode.ServiceInternalError
+
+
+def errors_handler(f) -> Callable:
+    @wraps(f)
+    def decorated(*args, **kwargs) -> ProgressEvent:
+        try:
+            return f(*args, **kwargs)
+        except Exception as e:
+            model = args[1].desiredResourceState or None
```

Comment:
This feels a bit dangerous in an exception handler? Maybe try/except that call?

Reply:
Fair point, i was thinking I can trust the underlying CF calls 😄